pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -285,11 +285,15 @@ func (c *Controller) syncProvisionRequestsPodTemplates(ctx context.Context, wl *
 			}
 
 			// apply the admission node selectors to the Template
-			selectors, err := c.getFlavorSelectors(ctx, psaMap[psName])
+			psi, err := podsetinfo.FromAssignment(ctx, c.client, psaMap[psName], reqPS.Count)
```

Comment:
Add at least one unit test case for annotations

Reply:
done